dcos/marathon.py

```diff
@@ -741,10 +743,30 @@ def remove_pod(self, pod_id, force=False):
         :rtype: None
         """
 
-        path = 'v2/pods' + _normalize_marathon_id_path(pod_id)
+        path = self._marathon_id_path_join('v2/pods', pod_id)
         params = {'force': 'true'} if force else None
         self._rpc.http_req(http.delete, path, params=params)
 
+    @staticmethod
+    def _marathon_id_path_join(url_path, id_path):
+        """Concatenates a URL path with a Marathon "ID path", ensuring the
+        result is well-formed.
+
+        The path and the ID will be joined with a single forward slash (/),
+        all trailing slashes in the ID will be removed, and the ID will have
```

Comment:
Ah, notice in #718 that you added the trailing slash to the Marathon base URL, which always has the endpoint path appended to it afterward anyway. Jan-Philip points out the source of the problem in DCOS-6961: it's the counterintuitive behavior of `urllib.parse.urljoin()`, which will drop the last segment of the first path argument:

```python
urllib.parse.urljoin('v2/pods', 'foo/bar') == 'v2/foo/bar' # !!!
urllib.parse.urljoin('v2/pods', '/foo/bar') == '/foo/bar'
urllib.parse.urljoin('v2/pods/', 'foo/bar') == 'v2/pods/foo/bar'
urllib.parse.urljoin('v2/pods/', '/foo/bar') == '/foo/bar'
```

So (1) it's fine to remove trailing slashes from the ID, because that wasn't the cause of DCOS-6961, and (2) the correct behavior will be preserved by this, because `posixpath.join()` is being used instead.

There's a few test cases in this PR which cover exactly this behavior, but if you think there's a gap I'm happy to add more :smile: